File: Flower.Data/Entities/Coupon.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Flower.Data.Entities
{
    public class Coupon
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(50)]
        public string Code { get; set; } = string.Empty;

        [MaxLength(500)]
        public string? Description { get; set; }

        public DiscountType DiscountType { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal DiscountValue { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal? MinimumOrderAmount { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal? MaximumDiscountAmount { get; set; }

        public int? UsageLimit { get; set; }

        public int UsedCount { get; set; }

        public int? UsagePerCustomer { get; set; }

        public int? CustomerId { get; set; }

        public DateTime? StartDate { get; set; }

        public DateTime? EndDate { get; set; }

        public bool IsPublic { get; set; }

        public bool IsActive { get; set; } = true;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime? UpdatedAt { get; set; }

        public virtual ICollection<CouponUsage>? Usages { get; set; }
    }
}

File: flower-admin.frontend/src/types/coupon.ts
export type DiscountType = 'Percentage' | 'FixedAmount'

export interface CouponDTO {
  id: number
  code: string
  description?: string
  discountType: DiscountType
  discountValue: number
  minimumOrderAmount?: number
  maximumDiscountAmount?: number
  usageLimit?: number
  usedCount: number
  usagePerCustomer?: number
  customerId?: number
  startDate?: string
  endDate?: string
  isPublic: boolean
  isActive: boolean
  createdAt: string
  updatedAt?: string
}

export interface CreateCouponDTO {
  code: string
  description?: string
  discountType: DiscountType
  discountValue: number
  minimumOrderAmount?: number
  maximumDiscountAmount?: number
  usageLimit?: number
  usagePerCustomer?: number
  customerId?: number
  startDate?: string
  endDate?: string
  isPublic?: boolean
  isActive?: boolean
}

export interface UpdateCouponDTO {
  id: number
  code: string
  description?: string
  discountType: DiscountType
  discountValue: number
  minimumOrderAmount?: number
  maximumDiscountAmount?: number
  usageLimit?: number
  usagePerCustomer?: number
  customerId?: number
  startDate?: string
  endDate?: string
  isPublic?: boolean
  isActive?: boolean
}

export interface CouponUsageDTO {
  id: number
  couponId: number
  customerId: number
  orderId: number
  discountAmount: number
  usedAt: string
  couponCode?: string
  customerName?: string
}
